src/orange/surf/detail/InvoluteSolver.hh

```diff
@@ -0,0 +1,323 @@
+ * ). The initial bounds can be determined by the set of: /f[ {0, beta - a,
+ * beta - a - pi, beta - a + pi, beta - a - 2pi, beta - a + 2pi ...} /f] Where
+ * \em beta is: \f[ beta = arctan(-v/u) \f]
+ */
+CELER_FUNCTION auto
+InvoluteSolver::operator()(Real3 const& pos,
+                           Real3 const& dir) const -> Intersections
+{
+    // Flatten pos and dir in xyz and uv respectively
+    real_type x = pos[0];
+    real_type const y = pos[1];
+    real_type const z = pos[2];
+
+    real_type u = dir[0];
+    real_type const v = dir[1];
+    real_type const w = dir[2];
+
+    if (mirror_)
+    {
+        x = -x;
+        u = -u;
+    }
+
+    // Lambda used for calculating the roots using Regular Falsi Iteration
+    auto root = [](real_type t,
+                   real_type x,
+                   real_type y,
+                   real_type u,
+                   real_type v,
+                   real_type r_b_,
+                   real_type a_) {
+        real_type a = u * std::sin(t + a_) - v * std::cos(t + a_);
+        real_type b = t * (u * std::cos(t + a_) + v * std::sin(t + a_));
+        real_type c = r_b_ * (a - b);
+        return c + x * v - y * u;
+    };
+    /*
+     * Results initalization
+     */
+    Intersections result;
+    // Initial result vector.
+    result = {no_intersection(), no_intersection(), no_intersection()};
+
+    // Return result if particle is travelling along z-axis.
+    if (u == 0 && v == 0)
+    {
+        return result;
+    }
+
+    // Initialize distance vector with root counter j
+    Array<real_type, 3> dist;
+    real_type j = 0;
+
+    // Conversion constant for 2-D distance to 3-D distance
+
+    real_type convert = std::sqrt(ipow<2>(v) + ipow<2>(u) + ipow<2>(w))
+                        / std::sqrt(ipow<2>(v) + ipow<2>(u));
+
+    /*
+     * Define tolerances.
+     * tol_point gives the tolerance level for a point and is set to 1e-7,
+     * account for the floating point error when performing square roots.
+     * tol_conv gives the tolerance for the Regular Falsi iteration,
+     * ensuring that the root found produces a result that within 1e-8 of 0.
+     */
+    real_type const tol_point = 1e-7;
+    real_type const tol_conv = 1e-8;
+
+    // Check if particle is on a surface within tolerance
+
+    real_type const rxy2 = ipow<2>(x) + ipow<2>(y);
+    real_type const t_point = std::sqrt((rxy2 / (ipow<2>(r_b_))) - 1);
+    real_type angle = t_point + a_;
+    real_type x_inv = r_b_ * (std::cos(angle) + t_point * std::sin(angle));
+    real_type y_inv = r_b_ * (std::sin(angle) - t_point * std::cos(angle));
+    if (std::fabs(x - x_inv) < tol_point && std::fabs(y - y_inv) < tol_point)
+    {
+        dist[j] = 0;
+        j++;
+    }
+
+    // Line angle parameter
+
+    real_type beta;
+
+    if (u != 0)
+    {
+        // Standard method
+        beta = std::atan(-v / u);
+    }  // Edge case
+    else if (-v < 0)
+    {
```

Comment:
Comments go inside of the block that they apply to. Also check the negation here:
```suggestion
    } 
    else if (v > 0)
    {
         // Edge case
```